### PR TITLE
ORT Multi-Interface Support

### PR DESCRIPTION
**Pull request by rob05c**
Adds support for Multiple Interfaces to ORT.

Includes getting the right interface from the Server, as well as setting the records.config `outgoing_ip` setting to the Service Interface(s).

This also refactors remap.config and parent.config, to use standard API objects. Those were already in the works and had a PR, and multi-interface support would cause huge conflicts, as well as much of the same code to review. A single PR both avoids conflicts and makes it less code to review.

No changelog, no interface change.
No docs, multi-interface servers is already documented.
Includes unit tests.

Manually tested, config generates the same before and after this change for our production servers, as well as our test environment with Topologies.

- [x] This PR is not related to any other Issue

## Which Traffic Control components are affected by this PR?
- Traffic Ops ORT

## What is the best way to verify this PR?
Run unit tests.
Generate config with ORT before this change, build this and generate new config, diff and verify there are no semantic changes, except the additional record.config entry.

## If this is a bug fix, what versions of Traffic Control are affected?
Not a bug fix.

## The following criteria are ALL met by this PR
- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

## Additional Information

### CODE REVIEW

File: lib/go-atscfg/ipallowdotconfig.go [hunk 1]
```diff
@@ -166,48 +170,79 @@ func MakeIPAllowDotConfig(
...
+			// - all children of this server
+			// - all monitors, if this server is a Mid
+			//
+			// TODO: handle Topologies. Mids currently block everything but Edges
+			//       We should decide how to handle that in a post-edge-mid world.
+			//       That probably means adding all child and monitor IPs, and blocking everything else,
+			//       for all non-first-tier caches.
+			//
+			_, isChild := childCGs[*childServer.Cachegroup]
+			if !isChild && (!strings.HasPrefix(server.Type, tc.MidTypePrefix) || string(childServer.Type) != tc.MonitorTypeName) {
```
[the remark this answers is not in the document]
Comment (rob05c) [reply lib/go-atscfg/ipallowdotconfig.go:217]: Not sure what you mean. The last group does have parenthesis around it, there's no operator precedence happening here...?

File: lib/go-atscfg/ipallowdotconfig.go [hunk 2]
```diff
@@ -166,48 +170,79 @@ func MakeIPAllowDotConfig(
...
+				continue
+			}
+			// 	childServers[tc.CacheName(*sv.HostName)] = atscfg.IPAllowServer{IPAddress: sv.IPAddress, IP6Address: sv.IP6Address}
```
[the remark this answers is not in the document]
Comment (rob05c) [reply lib/go-atscfg/ipallowdotconfig.go]: Yep, commented and forgot to remove. Removed.

File: lib/go-atscfg/ipallowdotconfig.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ezelkow1) [reply lib/go-atscfg/ipallowdotconfig.go:217]: I just mean like 
if !isChild && (!strings.HasPrefix(server.Type, tc.MidTypePrefix) || (string(childServer.Type) != tc.MonitorTypeName))

its more of just a personal preference, not a big deal but it makes it easier to see 2 separate blocks of operations separated by the or. Up to you if you want to do it or not

[the remark this answers is not in the document]
Comment (rob05c) [reply lib/go-atscfg/ipallowdotconfig.go]: Ah. I don't have a strong opinion either way. Changed.

**Pull request merged**